pkg/pod/entrypoint.go

```diff
@@ -209,7 +225,9 @@ func resultArgument(steps []corev1.Container, results []v1.TaskResult) []string
 func collectResultsName(results []v1.TaskResult) string {
 	var resultNames []string
 	for _, r := range results {
-		resultNames = append(resultNames, r.Name)
+		if r.Value == nil {
+			resultNames = append(resultNames, r.Name)
+		}
```

Comment:
what's this change used for?

Reply:
its to distinguish between results written to `results.resName.path` (i.e. auto surfaced) vs once that need a value from step results.